src/nvim/shada.c

```diff
@@ -1298,7 +1289,11 @@ static char *shada_filename(const char *file)
 {
   if (file == NULL || *file == NUL) {
     if (p_shadafile != NULL && *p_shadafile != NUL) {
-      file = p_shadafile;
+      if (!strequal(p_shadafile, "NONE")) {
```

Comment:
the old comment had useful information (calling out `--clean`).
```suggestion
      // Check if writing to shada file was disabled ("-i NONE" or "--clean").
      if (!strequal(p_shadafile, "NONE")) {
```